chain/beacon/chain.go

```diff
@@ -65,13 +78,10 @@ func (c *chainStore) NewValidPartial(addr string, p *drand.PartialBeaconPacket)
 	}
 }
 
-func (c *chainStore) NewBeacon(addr string, proto *drand.BeaconPacket) {
-	c.newBeaconCh <- protoToBeacon(proto)
-}
-
 func (c *chainStore) Stop() {
-	c.Store.Close()
+	c.CallbackStore.Close()
 	close(c.done)
+	c.CallbackStore.RemoveCallback("chainstore")
```

Comment:
is there a reason callbacks aren't removed by `Close`? I would expect this to either happen before calling `Close` or for `Close` to do this internally. Using objects after closing them indicates something hasn't actually stopped fully when calling Close, which is worrying

Reply:
No you're right :)